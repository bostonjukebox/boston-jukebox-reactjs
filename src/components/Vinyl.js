import React from 'react'
import PropTypes from 'prop-types'
import styled, { keyframes } from 'styled-components'

import source from '../img/vinyl.png'

const Container = styled.section`
    margin-top: 2em;
`

const spin = keyframes`
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
`

const Icon = styled.img`
    width: 100px;
    height: 100px;
`

const IconAnimated = Icon.extend`
    animation: ${spin} 4s linear infinite;
`

const Vinyl = ({ playing }) =>
  <Container>
    {(playing) ?
      <IconAnimated src={source} alt="Vinyl" />
      : <Icon src={source} alt="Vinyl" />
    }
  </Container>

Vinyl.propTypes = {
  playing: PropTypes.bool,
}

export default Vinyl
