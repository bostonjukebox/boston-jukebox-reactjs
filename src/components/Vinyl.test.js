import React from 'react'
import renderer from 'react-test-renderer'
import Vinyl from './Vinyl'

test('renders without crashing', () => {
    const tree = renderer.create(
        <Vinyl/> ).toJSON();
    expect(tree).toMatchSnapshot();
}); 