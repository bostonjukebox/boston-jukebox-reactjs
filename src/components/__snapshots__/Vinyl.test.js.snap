// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders well 1`] = `
<section
  className="sc-bdVaJa brgSrh"
>
  <img
    alt="Vinyl"
    className="sc-bwzfXH cxTssG"
    src="vinyl.png"
  />
</section>
`;

exports[`renders without crashing 1`] = `
<section
  className="sc-bdVaJa brgSrh"
>
  <img
    alt="Vinyl"
    className="sc-bwzfXH cxTssG"
    src="vinyl.png"
  />
</section>
`;
